listener.go

```diff
@@ -10,27 +10,11 @@ import (
 
 // Listen creates a DTLS listener
 func Listen(network string, laddr *net.UDPAddr, config *Config) (net.Listener, error) {
-	if err := validateConfig(config); err != nil {
-		return nil, err
-	}
-
-	lc := udp.ListenConfig{
-		AcceptFilter: func(packet []byte) bool {
-			pkts, err := recordlayer.UnpackDatagram(packet)
-			if err != nil || len(pkts) < 1 {
-				return false
-			}
-			h := &recordlayer.Header{}
-			if err := h.Unmarshal(pkts[0]); err != nil {
-				return false
-			}
-			return h.ContentType == protocol.ContentTypeHandshake
-		},
-	}
-	parent, err := lc.Listen(network, laddr)
+	parent, err := checkConfigAndListenUDP(network, laddr, config)
```

Comment:
We want to reuse this code and notice if there has been changes to it when merging